controllers/databaseengine_controller.go

```diff
@@ -331,18 +337,60 @@ func (r *DatabaseEngineReconciler) listPendingOperatorUpgrades(
 			}
 			// Check if the version is greater than the current version.
 			if semver.Compare("v"+version, "v"+currentVersion) > 0 {
-				result = append(result, everestv1alpha1.OperatorUpgrade{
-					TargetVersion: version,
-					InstallPlanRef: corev1.LocalObjectReference{
-						Name: ip.GetName(),
-					},
-				})
+				if _, ok := result[version]; ok {
+					continue // we already have this version, with the newest IP
+				}
+				result[version] = ip.GetName()
 			}
 		}
 	}
+	return result
+}
+
+func (r *DatabaseEngineReconciler) listPendingOperatorUpgrades(
+	ctx context.Context,
+	dbEngine *everestv1alpha1.DatabaseEngine,
+) ([]everestv1alpha1.OperatorUpgrade, error) {
+	// If OLM is not installed, we cannot check for pending upgrades.
+	if !r.isOLMInstalled(ctx) {
+		return nil, nil
+	}
+	// We need some version to be reported first.
+	currentVersion := dbEngine.Status.OperatorVersion
+	if currentVersion == "" {
+		return nil, nil
+	}
+
+	// Get the Subscription for this operator.
+	subscription := &opfwv1alpha1.Subscription{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Name:      dbEngine.GetName(),
+		Namespace: dbEngine.GetNamespace(),
+	}, subscription); err != nil {
+		return nil, err
+	}
+
+	// List install plans in this namespace.
+	installPlans := &opfwv1alpha1.InstallPlanList{}
+	if err := r.List(ctx, installPlans, client.InNamespace(dbEngine.GetNamespace())); err != nil {
+		return nil, err
+	}
+
+	installPlanRefs := getInstallPlanRefsForUpgrade(dbEngine, subscription, installPlans)
+	result := []everestv1alpha1.OperatorUpgrade{}
+	for v, ipName := range installPlanRefs {
+		result = append(result, everestv1alpha1.OperatorUpgrade{
+			TargetVersion:  v,
+			InstallPlanRef: corev1.LocalObjectReference{Name: ipName},
+		})
+	}
 	return result, nil
 }
 
+// get the status and version of the operator deployment.
+// TODO: can we instead check the status of the corresponding CSV?
```

Comment:
I'm not sure either :confused:

Reply:
I think this is possible, but then it will stop working with our Tilt setup because we don't use OLM there.
Once we have a Helm chart, I think we can simplify our tilt dev setup too..